src/subcommand/fix.rs

```diff
@@ -0,0 +1,484 @@
+    CheckIntegration {
+        /// the remote that the integration check failed for
+        remote: Remote,
+        /// the error returned by the integration check
+        error: String,
+        /// A message explaining how to fix this error
+        msg: String,
+    },
+
+    /// Make a GET request to a fossa endpoint that does not require authentication
+    #[error("check fossa connection: {}", msg)]
+    CheckFossaGet {
+        /// An error message explaining how to fix this GET from FOSSA
+        msg: String,
+    },
+
+    /// Creating a full URL from the provided endpoint
+    #[error("create FOSSA URL from endpoint {}/{}", remote, path)]
+    CreateFullFossaUrl {
+        /// The configured fossa URL
+        remote: url::Url,
+        /// The path on Fossa
+        path: String,
+    },
+}
+
+impl Error {
+    #[tracing::instrument]
+    fn fix_explanation(&self) -> String {
+        match self {
+            Error::CheckIntegration { remote, msg, .. } => {
+                format!("❌ {}\n\n{}", remote.to_string().red(), msg)
+            }
+            Error::CheckFossaGet { msg } => {
+                format!("❌ {} {}", "Error checking connection to FOSSA:".red(), msg)
+            }
+            Error::CreateFullFossaUrl { remote, path } => {
+                format!(
+                    "❌ Creating a full URL from your remote of {} and path = {}",
+                    remote, path
+                )
+            }
+        }
+    }
+
+    #[tracing::instrument]
+    fn integration_error(
+        remote: &Remote,
+        transport: &Transport,
+        err: Report<repository::Error>,
+    ) -> Self {
+        let msg = formatdoc!(
+            "
+            We encountered an error while trying to connect to your git remote at {}.
+
+            {}
+
+            Full error message from git:
+
+            {}
+
+            ",
+            remote,
+            Self::integration_connection_explanation(transport),
+            err.to_string(),
+        );
+        Error::CheckIntegration {
+            remote: remote.clone(),
+            error: err.to_string(),
+            msg,
+        }
+    }
+
+    #[tracing::instrument]
+    fn integration_connection_explanation(transport: &transport::Transport) -> String {
+        let shared_instructions = "We were unable to connect to this repository. Please make sure that the authentication info and the remote are set correctly in your config.yml file.";
+        let base64_command = r#"echo -n "<username>:<password>" | base64"#.green();
+        let specific_instructions = match transport {
+            transport::Transport::Ssh {
+                auth: ssh::Auth::KeyFile(key_path),
+                endpoint,
+            } => {
+                let key_path = key_path.to_string_lossy();
+                let command = format!(
+                    r#"GIT_SSH_COMMAND="ssh -i {} -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -F /dev/null" git ls-remote {}"#,
```

Comment:
Is it (relatively) easy to use functions in `repository` to get the actual args used for these commands so these are always up to date?

If so please strongly consider doing so, I would hate for this to be silently out of date with what we're actually doing. If not, please add tests to validate the args match what we're logging here in your testing PR, so we're notified if they do get out of date.

Reply:
That's a realy good point, and I think I've come up with a reasonable solution for this. It makes the example commands in the `broker fix` output a little longer, but that's because they're now an exact copy of the commands we're running.

Also, the example commands won't work as-is on Windows, but they wouldn't in the previous implementation either. I'm going to make a ticket to fix this later unless you object

Reply:
Yeah, ticket for later sgtm